mne/simulation/metrics.py

```diff
@@ -63,3 +67,310 @@ def source_estimate_quantification(stc1, stc2, metric='rms'):
         score = 1. - (np.dot(data1.flatten(), data2.flatten()) /
                       (norm(data1) * norm(data2)))
     return score
+
+
+def _uniform_stc(stc1, stc2):
+    """Uniform vertices of two stcs.
+
+    This function returns the stcs with the same vertices by
+    inserting zeros in data for missing vertices.
+    """
+    if len(stc1.vertices) != len(stc2.vertices):
+        raise ValueError('Data in stcs must have the same number of vertices '
+                         'components. Got %d != %d.' %
+                         (len(stc1.vertices), len(stc2.vertices)))
+    idx_start1 = 0
+    idx_start2 = 0
+    stc1 = stc1.copy()
+    stc2 = stc2.copy()
+    all_data1 = []
+    all_data2 = []
+    for i, (vert1, vert2) in enumerate(zip(stc1.vertices, stc2.vertices)):
+        vert = np.union1d(vert1, vert2)
+        data1 = np.zeros([len(vert), stc1.data.shape[1]])
+        data2 = np.zeros([len(vert), stc2.data.shape[1]])
+        data1[np.searchsorted(vert, vert1)] = \
+            stc1.data[idx_start1:idx_start1 + len(vert1)]
+        data2[np.searchsorted(vert, vert2)] = \
+            stc2.data[idx_start2:idx_start2 + len(vert2)]
+        idx_start1 += len(vert1)
+        idx_start2 += len(vert2)
+        stc1.vertices[i] = vert
+        stc2.vertices[i] = vert
+        all_data1.append(data1)
+        all_data2.append(data2)
+
+    stc1._data = np.concatenate(all_data1, axis=0)
+    stc2._data = np.concatenate(all_data2, axis=0)
+    return stc1, stc2
+
+
+def _apply(func, stc_true, stc_est, per_sample):
+    """Apply metric to stcs.
+
+    Applies a metric to each pair of columns of P and Q
+    if per_sample is True. Otherwise it applies it to P and Q
+    directly.
+    """
+    if per_sample:
+        metric = np.zeros(stc_true.data.shape[1])  # one value per time point
```

Comment:
I would use np.empty not np.zeros (more explicit that the values will be overwritten)